target/linux/ipq40xx/image/Makefile

```diff
@@ -221,6 +221,30 @@ define Device/qcom_ap-dk04.1-c1
 endef
 TARGET_DEVICES += qcom_ap-dk04.1-c1
 
+define Device/qxwlan_e2600ac-c1
+	$(call Device/FitImage)
+	DEVICE_TITLE :=  Qxwlan E2600AC-C1 (nor flash)
+	BOARD_NAME := e2600ac-c1
+	DEVICE_DTS := qcom-ipq4019-qxwlan-e2600ac-c1
+	KERNEL_SIZE := 4096k
+	IMAGE_SIZE := 26624k
+	IMAGES := sysupgrade.bin
+	IMAGE/sysupgrade.bin := append-kernel |append-rootfs | pad-rootfs | append-metadata
+endef
+TARGET_DEVICES += qxwlan_e2600ac-c1
+
+define Device/qxwlan_e2600ac-c2
+	$(call Device/FitImage)
+	$(call Device/UbiFit)
+	DEVICE_DTS := qcom-ipq4019-qxwlan-e2600ac-c2
+	KERNEL_INSTALL := 1
+	KERNEL_SIZE := 4096k
```

Comment:
Is this hard limit 4MiB due to some u-boot code? Because the QSDK's u-boot ubi support reads the whole volume so the kernel size shouldn't matter?!

Hence, can you please provide the full u-boot env (printenv) for your device? Because this should be fixable.

Reply:
Hi，
May I still need to modify it?

Reply:
(IPQ40xx) # printenv
baudrate=115200
bootcmd=bootipq
bootdelay=2
ethact=eth0
flash_type=0
ipaddr=192.168.1.1
lelf=tftpb 0x84000000 ${dir}openwrt-ipq40xx-u-boot-stripped.elf&&sf probe&&sf erase 0xf0000 0x80000&&sf write 0x84000000 0xf0000 $filesize
lfw=tftpb 0x84000000 ${dir}ubi-firmware.bin&&&&nand erase 0x0 0x4000000&&nand write 0x84000000 0x0 $filesize
lu=tftpb 0x84000000 ${dir}u-boot.bin&&sf probe&&sf erase 0x0000 0x170000&&sf write 0x84000000 0x0000 $filesize
machid=8010206
serverip=192.168.1.10
stderr=serial
stdin=serial
stdout=serial

Environment size: 554/65532 bytes